### PR TITLE
[#18] Refactor and remove several performance heavy parts of code

### PR DESCRIPTION
**Pull request by svetozarsadecky**
#18 - This is an attempt to resolve performance issues;

I have removed whole Vue instance that was responsible for reactivity in currentPopups.
Also removed duplicate code in popup component that was directly mutating currentPopups attribute.
 
What can and definitely will help in nuxt, is declaring the popup for client side only.

### CODE REVIEW

File: src/components/Popup.vue [hunk 1]
```diff
@@ -233,13 +228,13 @@ export default {
...
+                this.onClose();
             }
         },
     },
 };
 </script>
 
 <style lang="less">
-@import '../less/popup.less';
+    @import '../less/popup.less';
```
Comment (LordZombi) [inline src/components/Popup.vue:239]: cmon

File: src/main.js [hunk 2]
```diff
@@ -4,7 +4,6 @@ import popup from './plugin/popup';
 import App from './App';
 
 Vue.config.productionTip = false;
```
Comment (LordZombi) [inline src/main.js:6]: totok treba ?

File: src/plugin/popup.js [hunk 3]
```diff
@@ -6,67 +6,46 @@ const Plugin = {
...
-        /**
-         * Create getters/setters for reactive properties
-         */
         Object.defineProperties(Vue.prototype.$popup, {
-            currentPopups: {
-                get: () => this.currentPopups.getCurrentPopups,
+            show: {
+                value: (name) => {
+                    if (typeof name === 'string') {
+                        Vue.prototype.$popup.currentPopups = [...currentPopups.add(name)];
```
Comment (LordZombi) [inline src/plugin/popup.js:29]: nevadi ze pred tym to bol Set a ted je to Array?
bude do spatne kompatobilne?

File: src/components/Popup.vue [hunk 4]
```diff
@@ -14,15 +14,15 @@
         >
             <div
                 class="popup__close"
-                @click.prevent="close"
+                @click.prevent="onClose"
```
Comment (LordZombi) [inline src/components/Popup.vue:17]: wtf ? preco si to zmenil ved pred tym to robilo nieco ine ? 

close - emitlo aj `close` event aj zavolalo `closePopup` co neviem co robi ale asi nieco dolezite
onClose - len hide-ne popup, nezavola sa ani event ... ?

File: src/main.js (hunk 2 above)
Comment (svetozarsadecky) [reply src/main.js:6]: Len tym vypnes log:
```
You are running Vue in development mode.
Make sure to turn on production mode when deploying for production.
See more tips at https://vuejs.org/guide/deployment.html
```

File: src/plugin/popup.js (hunk 3 above)
Comment (svetozarsadecky) [reply src/plugin/popup.js:29]: Nemalo by. Na strane usera nepotrebujes Set. Toto ti je treba interne pre ucely pluginu.

File: src/components/Popup.vue (hunk 1 above)
Comment (svetozarsadecky) [reply src/components/Popup.vue]: Eslint rulz predpokladam v php storme.

**Pull request closed without merging**